apps/challenges/aws_utils.py

```diff
@@ -452,23 +661,45 @@ def register_task_def_by_challenge_pk(client, queue_name, challenge):
                 return e.response
             # challenge host auth token to be used by code-upload-worker
             token = JwtToken.objects.get(user=challenge.creator.created_by)
-            definition = task_definition_code_upload_worker.format(
-                queue_name=queue_name,
-                container_name=container_name,
-                ENV=ENV,
-                challenge_pk=challenge.pk,
-                auth_token=token.refresh_token,
-                cluster_name=cluster_name,
-                cluster_endpoint=cluster_endpoint,
-                certificate=cluster_certificate,
-                CPU=worker_cpu_cores,
-                MEMORY=worker_memory,
-                log_group_name=log_group_name,
-                EVALAI_DNS=EVALAI_DNS,
-                EFS_ID=efs_id,
-                **COMMON_SETTINGS_DICT,
-                **challenge_aws_keys,
-            )
+            if challenge.is_static_dataset_docker_based_challenge:
+                definition = task_definition_static_code_upload_worker.format(
```

Comment:
If we are using `code_upload_worker` task def we won't need a separate config here. The cluster creation setup for static code upload challenge is exactly same as code upload worker.

Reply:
@Ram81 Yeah, `containerDefinitions` parameter is a list and contains the container info. If we wish to use same defn for both code upload and static code upload challenge, we might need to contain {containerDefinitions} as a variable in the code upload task defn, currently, it is provided as-is. So in that case for static code upload, we would create containerDefinitions with two containers and in case of code upload pass only one container. This might be a workaround but limits the flexibility.

Reply:
I see, if that is the case then we can have separate task def for both challenge types